src/sagemaker/model.py

```diff
@@ -407,6 +408,9 @@ def _compilation_job_config(
             else input_shape,
             "Framework": framework.upper(),
         }
+        if framework.upper() == "PYTORCH" and framework_version is not None:
```

Comment:
Please add a check for target_device is a cloud ioc target. Framework version is not supported for edge devices and ml_inf1.

Reply:
Addressed. Thanks!!